packages/gatsby/src/redux/nodes.ts

```diff
@@ -3,9 +3,23 @@ import { IGatsbyNode } from "./types"
 import { createPageDependency } from "./actions/add-page-dependency"
 import { IDbQueryElemMatch } from "../db/common/query"
 
+// Only list supported ops here. "CacheableFilterOp"
+type FilterOp = "$eq" | "$lte"
+type FilterValue = string | number | boolean
```

Comment:
As far as I know the values are only ever str/num/bool and a hint of null.

@vladar or @freiksenet can answer this better.

This approach would later be used for `ne` and `in` as well so it's good to know regardless.

Reply:
I did some checks and filter value is always a `string`. Even for dates. So it works fine when the node has a date field expressed as a `string`.

But there is a bug when a node field contains an instance of `Date`. Sift filtering is broken for this situation (as we compare string vs date). At least I tried it and it didn't work.

The proper fix would be to convert the filter value to `Date` when running sift ([docs](https://github.com/crcn/sift.js#date-comparison)).

An alternative option would be to always store dates as strings in nodes (maybe a better one). But this may be a breaking change.

Reply:
Added this PR with a failing test case (skipped for now): https://github.com/gatsbyjs/gatsby/pull/23096

We should address it in v3 (as it is a breaking change). It will also ensure that filtering with indexes is consistent with sift. 

Also, in the worst case, it will just fallback to querying without index so I think this shouldn't be a blocker for this PR.